### PR TITLE
unwrap PSObject in PowerShellKernel.TryGetVariable

### PR DESCRIPTION
**Pull request by jonsequitur**

### CODE REVIEW

File: Microsoft.DotNet.Interactive.Formatting/JsonFormatter.cs [hunk 1]
```diff
@@ -25,7 +25,7 @@ static JsonFormatter()
 
         internal static IFormatterSet DefaultFormatters { get; private set; } = new DefaultHtmlFormatterSet();
     
-        internal static JsonSerializerSettings SerializerSettings { get; }
+        public static JsonSerializerSettings SerializerSettings { get; }
```
Comment (TylerLeonhardt) [inline Microsoft.DotNet.Interactive.Formatting/JsonFormatter.cs]: How strict are you about documenting public properties?

You use this in another project, so I would think it'd need some doc comments.

File: Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs [hunk 2]
```diff
@@ -106,10 +106,19 @@ private PowerShell CreatePowerShell()
         public override bool TryGetVariable(string name, out object value)
         {
             var variable = _lazyPwsh.Value.Runspace.SessionStateProxy.PSVariable.Get(name);
-            
+
             if (variable != null)
             {
-                value = variable.Value;
+                switch (variable.Value)
```
Comment (TylerLeonhardt) [inline Microsoft.DotNet.Interactive.PowerShell/PowerShellKernel.cs]: Didn't want to use the pattern matching switch syntax?

File: Microsoft.DotNet.Interactive.Formatting/JsonFormatter.cs (hunk 1 above)
Comment (jonsequitur) [reply Microsoft.DotNet.Interactive.Formatting/JsonFormatter.cs]: We'll do a documentation pass before too long but method signatures might change or get refactored so I prefer to let them stabilize a little first.

**Comment by daxian-dbw** [comment]
I opened https://github.com/dotnet/interactive/issues/246 to track revisiting the implementation of `TryGetVariable` in PS kernel.

**Pull request merged**